src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/InterchangeHeader.java

```diff
@@ -66,30 +59,48 @@ public String getKey() {
     }
 
     @Override
-    public String getValue() {
-        String timestamp = DATE_FORMAT.format(translationTime);
-        String formattedSequenceNumber = String.format("%08d", sequenceNumber);
-        return "UNOA:2" + "+" + sender + "+" + recipient + "+" + timestamp + "+" + formattedSequenceNumber;
-    }
-
-    @Override
-    protected void validateStateful() throws EdifactValidationException {
-        if (sequenceNumber == null) {
+    public void validate() throws EdifactValidationException {
+        if (StringUtils.isBlank(getSender())) {
```

Comment:
Maybe a switch statement will look a bit better in this case because there are 5 if statements, but it's just a suggestion and can be ignored

Reply:
Can you give an example on how switch could be applied here? I think it can't

Reply:
Yea, you a right, will not make sense in this case. Sorry, my bad